### DISCUSSION TITLE
API Returns Incomplete Data in Extended Response

### DISCUSSION THREAD

**Discussion by jilliansd** [amzn/ads-advanced-tools-docs#43] [General]
We are exporting campaign data manually for SP campaigns from AMS dashboard and then fetching the same data via api using the following endpoint:

https://advertising-api.amazon.com/v2/sp/campaigns/extended

The response data received via API does not have the following information or field:  “**Avg. time in budget**”, we are however getting this field when we export the data manually from the AMS portal.

Please advise if there is any other api endpoint/report that can provide the information for “Avg. time in budget” data for a campaign.

Not able to find any documentation supporting this information, if the data is available via manual export then similar data/fields should be available for the api as well.

**Comment by elizafarley** [comment]
Hi @jilliansd, I believe the data you are talking about (including Avg. time in budget) comes from the SP budget report in the UI. The Unfortunately the SP budget report is [not available](https://advertising.amazon.com/API/docs/en-us/reporting/v2/advertising-console#unsupported-report-types) in the API. If this feature is supported in the future, we will post in the [release notes](https://advertising.amazon.com/API/docs/en-us/info/release-notes). 

If you want to use the API to get more information on budget usage, you can try the [Budget recommendations and missed opportunities ](https://advertising.amazon.com/API/docs/en-us/sponsored-products/budget-recommendations-and-missed-opportunities) and [Budget usage](https://advertising.amazon.com/API/docs/en-us/sponsored-products/budget-usage/overview) endpoints.

**Comment by jilliansd** [comment]
Hi Eliza,

Thank you for your prompt response to the query, we were actually looking
for an endpoint that could give us the reporting data for the campaigns
budget. Hoping that this is made available in the near future - this report
is available via the UI for manual exports.

On Tue, Jan 10, 2023 at 9:33 PM Eliza Farley ***@***.***>
wrote:

> Hi @jilliansd <https://github.com/jilliansd>, I believe the data you are
> talking about (including Avg. time in budget) comes from the SP budget
> report in the UI. The Unfortunately the SP budget report is not available
> <https://advertising.amazon.com/API/docs/en-us/reporting/v2/advertising-console#unsupported-report-types>
> in the API. If this feature is supported in the future, we will post in the release
> notes <https://advertising.amazon.com/API/docs/en-us/info/release-notes>.
>
> If you want to use the API to get more information on budget usage, you
> can try the Budget recommendations and missed opportunities
> <https://advertising.amazon.com/API/docs/en-us/sponsored-products/budget-recommendations-and-missed-opportunities>
> and Budget usage
> <https://advertising.amazon.com/API/docs/en-us/sponsored-products/budget-usage/overview>
> endpoints.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/amzn/ads-advanced-tools-docs/discussions/43#discussioncomment-4647024>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AS2PIQM6OID4AZBYYIKEN2TWRWB6BANCNFSM6AAAAAATVTK7FA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


-- 
Regards,
Jillian

Jillian Macgowan
***@***.***    |    www.salesduo.com
